web/views/attendance_view.ex

```diff
@@ -2,8 +2,9 @@ defmodule CoursePlanner.AttendanceView do
   use CoursePlanner.Web, :view
 
   def get_teacher_display_name(offered_course_teachers) do
-    Enum.reduce(offered_course_teachers, "", fn(teacher, out) ->
-              Enum.join([out, teacher.name, teacher.family_name], " ") end)
+    offered_course_teachers
+    |> Enum.map(fn(teacher) -> Enum.join([teacher.name, teacher.family_name], " ") end)
```

Comment:
there is already a function name get_student_display_name. you can make it generic like "get_user_display_name" and use it for both

PS : in we have to modify the function to include the nick name if it's not "nil" too. cause in some places we now display the nickname and in some places we don't